kqueen/engines/eks.py

```diff
@@ -0,0 +1,220 @@
+from kqueen.config import current_config
+from kqueen.engines.base import BaseEngine
+import boto3
+
+import logging
+import yaml
+
+logger = logging.getLogger('kqueen_api')
+config = current_config()
+
+STATE_MAP = {
+    'CREATING': config.get('CLUSTER_PROVISIONING_STATE'),
+    'ACTIVE': config.get('CLUSTER_OK_STATE'),
+    'DELETING': config.get('CLUSTER_DEPROVISIONING_STATE'),
+    'FAILED': config.get('CLUSTER_ERROR_STATE'),
+    'UPDATED': config.get('CLUSTER_UPDATING_STATE')
+}
+
+
+class EksEngine(BaseEngine):
+    """
+    Amazon Elastic Kubernetes Service
+    """
+    name = 'eks'
+    verbose_name = 'Amazon Elastic Kubernetes Service'
+    parameter_schema = {
+        'provisioner': {
+            'aws_access_key': {
+                'type': 'text',
+                'label': 'AWS Access Key',
+                'order': 0,
+                'validators': {
+                    'required': True
+                }
+            },
+            'aws_secret_key': {
+                'type': 'text',
+                'label': 'AWS Secret Key',
+                'order': 1,
+                'validators': {
+                    'required': True
+                }
+            }
+        },
+        'cluster': {
+            'node_count': {
+                'type': 'integer',
+                'label': 'Node Count',
+                'default': 3,
+                'validators': {
+                    'required': True,
+                    'min': 1,
+                    'number': True
+                }
+            },
+            'roleArn': {
+                'type': 'text',
+                'label': 'IAM Role ARN',
+                'validators': {
+                    'required': True
+                }
+            },
+            'subnetid': {
+                'type': 'text',
+                'label': 'Subnet Id',
```

Comment:
Subnet/s Id - since more than one can be chosen. Ideally we need to get list of them and provide ability to pick subnets on the fly, but for now we need to parse string to the list at least (on the UI part)